src/bootstrap.rs

```diff
@@ -35,24 +35,26 @@ pub fn start() {
 mod tests {
     use crate::test_utils::new_random_qp2p_for_unit_test;
     use crate::{Builder, Config, Event};
-    use env_logger;
     use std::collections::HashSet;
     use std::net::{IpAddr, Ipv4Addr};
     use std::sync::mpsc;
+    use std::thread;
+    use std::time::Duration;
 
     // Try to bootstrap to 4 different peer at once. Use an artificial delay for 3 of the peers.
     // Make sure in the end we have only one bootstrapped connection (all other connections are
     // supposed to be dropped).
     #[test]
     fn bootstrap_to_only_one_node() {
-        env_logger::init();
+        const NUM_CONNS: u64 = 4;
+        const DELAY_MS: u64 = 10;
 
         let mut bs_nodes = Vec::new();
```

Comment:
Normally since this is a test case, i woulnd't have put the review, but i don't want to merge PR with some of the commits saying "review fixes" so this needs rebased (or squashed) anyway so was thinking i'll just mention this review as well if you want to incorporate that:
Since now you have constants for `NUM_CONNS` etc. which is good, why not also do `with_capacity()` for the containers which allow it ? Not critical for test cases but if other changes are there the we can just do these too.